modules/mergeImageWithTelemetry/mergeImageWithTelemetry.py

```diff
@@ -0,0 +1,84 @@
+from modules.mergeImageWithTelemetry.mergedData import MergedData
+from modules.timestamp.timestamp import Timestamp
+
+import datetime
+import numpy.typing as npt
+import typing
+
+class MergeImageWithTelemetry:
+    """
+    Merges image pipeline with telemetry pipeline
+    ...
+
+    Attributes
+    ----------
+    curData : Timestamp 
+        the current telemetry data 
+    start : bool
+        whether curData is None or not (whether get_closest_telemetry has been called before)
+    telemetryData : list<Timestamp>
+        list of the current telemetry data and their times in the form of Timestamp objects
+
+    Methods
+    -------
+    __init__()
+        initiallizes attributes
+    put_back(newTelemetryData: Timestamp)
+        places given telemetry data at the back of the telemetryData list
+    merge_with_closest_telemetry(imageTimestamp: datetime.datetime)
+        finds the telemetry with timestamp closest to the input datetime
+    """
+
+    def __init__(self): 
+        self.curData = None
+        self.start = True
+        self.telemetryData = []
+
+    def put_back(self, newTelemetryData: typing.type[Timestamp]):
+        self.telemetryData.append(newTelemetryData)
+
+    def merge_with_closest_telemetry(self, imageTimestamp: datetime.datetime, imageData : npt.ArrayLike ): 
+        """
+        finds the telemetry with timestamp closest to the input datetime
+        assumes that TelemetryData is sorted by increasing timestamp
+
+        Parameters 
+        ----------
+        imageTimestamp : datetime.datetime 
+            The time that the telemetry data should match with
+
+        Returns 
+        -------
+        [success, telemetry]
+
+        success : bool 
+            whether such a telemetry timestamp could be found
+        telemetry : Timestamp
+            the closest time and the corresponding telemetry data in a Timestamp object
+        """
+
+        if len(self.telemetryData) == 0:
+            if self.start: 
+                return False, None
+            return True, self.curData
```

Comment:
There is the case that the old image is matched, no new telemetry data comes in, and then a new image comes in and is matched with the exact same telemetry. What should probably happen is to hold onto the image and try to get new telemetry data instead.

Of course, this case should never happen (otherwise something has gone terribly wrong), but it's still possible (maybe the plane experiences a hiccup).